docs/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md

```diff
@@ -250,65 +252,426 @@ access_entries = {
+```
+
+A **Kubernetes job** will connects to the database and creates the necessary users with the required privileges. The script installs the necessary dependencies and runs SQL commands to create the IRSA user and assign it the correct roles and privileges.
+
+<Tabs groupId="env">
+  <TabItem value="standard" label="Standard" default>
+
+2. **Create a secret that references the environment variables**:
+
+```bash reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6/procedure/create-setup-db-secret.sh
+```
+
+This command creates a secret named `setup-db-secret` and dynamically populates it with the values from your environment variables.
+
+After running the above command, you can verify that the secret was created successfully by using:
+
+```bash
+kubectl get secret setup-db-secret -o yaml --namespace camunda
+```
+
+This should display the secret with the base64 encoded values.
+
+3. **Create a copy of the manifest**: Save the above manifest to a file, for example, `setup-postgres-create-db.yml`.
+
+```yaml reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6/setup-postgres-create-db.yml
+```
+
+  </TabItem>
+  <TabItem value="irsa" label="IRSA">
+
+2. **Create a secret that references the environment variables**:
+
+```bash reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6-irsa/procedure/create-setup-db-secret.sh
+```
+
+This command creates a secret named `setup-db-secret` and dynamically populates it with the values from your environment variables.
+
+After running the above command, you can verify that the secret was created successfully by using:
+
+```bash
+kubectl get secret setup-db-secret -o yaml --namespace camunda
+```
+
+This should display the secret with the base64 encoded values.
+
+3. **Create a copy of the manifest**: Save the above manifest to a file, for example, `setup-postgres-create-db.yml`.
+
+```yaml reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6-irsa/setup-postgres-create-db.yml
+```
+
+  </TabItem>
+</Tabs>
+
+3. **Apply the manifest**: Once the secret is created, the **Job** manifest from the previous step can consume this secret to securely access the database credentials.
+
+```bash
+kubectl apply -f setup-postgres-create-db.yml --namespace camunda
+```
+
+4. **Verify the job's completion**: Once the job is created, you can monitor its progress using:
+
+```bash
+kubectl get job/create-setup-user-db --namespace camunda --watch
+```
+
+Once the job shows as `Completed`, the users and databases will have been successfully created.
+
+5. **Check logs for confirmation**: You can view the logs of the job to confirm that the users were created and privileges were granted successfully:
+
+```bash
+kubectl logs job/create-setup-user-db --namespace camunda
```

Comment:
```
Create keycloak user and associated database
ERROR:  database "camunda_keycloak" already exists
CREATE ROLE
GRANT
GRANT
GRANT ROLE
```

as discussed in the PM, we should mention in the script that it's expected or something.